cleanlab/regression/rank.py

```diff
@@ -0,0 +1,153 @@
+import numpy as np
+from cleanlab.outlier import OutOfDistribution
+from sklearn.neighbors import NearestNeighbors
+from cleanlab.internal.regression_utils import assert_valid_inputs
+
+""" Generates label quality scores for every sample in regression dataset """
+
+
+def get_label_quality_scores(
+    labels: np.ndarray,
+    predictions: np.ndarray,
+    *,
+    method: str = "TO_BE_NAMED",  # TODO update name once finalised
+) -> np.ndarray:
+    """
+    Returns label quality score for each example in the regression dataset.
+
+    Each score is continous value in range [0,1]
+    1 - clean label (given label is likely correct).
+    0 - dirty label (given label is likely incorrect).
+
+    Parameters
+    ----------
+    labels : np.ndarray
+        Raw labels from original dataset.
+        Array of shape ``(N, )`` consisting given labels, where N is number of datapoints in the regression dataset.
+
+    predictions : np.ndarray
+        Predicated labels from regressor fitted on the dataset.
+        Array of shape ``(N,)`` consisting predicted labels, where N is number of datapoints in the regression dataset.
+
+    method : {"residual", "TO_BE_NAMED"}, default="residual" #TODO - update name once finalised
+
+    Returns
+    -------
+    label_quality_scores:
+        Array of shape ``(N, )`` of scores between 0 and 1, one per datapoint in the dataset.
+
+        Lower scores indicate datapoint more likely to contain a label issue.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from cleanlab.regression.rank import get_label_quality_scores
+    >>> labels = np.array([1,2,3,4])
+    >>> predictions = np.array([2,2,5,4.1])
+    >>> label_quality_scores = get_label_quality_scores(labels, predictions)
+    >>> label_quality_scores
+    array([0.36787944, 1.        , 0.13533528, 0.90483742])
+    """
+
+    # Check if inputs are valid
+    assert_valid_inputs(labels=labels, predictions=predictions, method=method)
+
+    scoring_funcs = {
```

Comment:
This annotation should fix the [mypy-error in the CI](https://github.com/cleanlab/cleanlab/actions/runs/3768279796/jobs/6406583011#step:5:25):

```suggestion
    scoring_funcs: Dict[str, Callable[[np.ndarray, np.ndarray], np.ndarray]] = {
```

Reply:
fixed in batch [f2c5862](https://github.com/cleanlab/cleanlab/pull/572/commits/f2c5862864c76f72113a244e7dc2320c1b574d28)